src/abi/tests.rs

```diff
@@ -0,0 +1,1478 @@
+                ty: IdlType::U64
+            },
+        ]
+    );
+
+    assert_eq!(idl.types.len(), 1);
+    assert_eq!(
+        idl.types[0],
+        IdlTypeDefinition {
+            name: "getNum_returns".to_string(),
+            docs: Some(vec![
+                "Data structure to hold the multiple returns of function getNum".to_string()
+            ]),
+            ty: IdlTypeDefinitionTy::Struct {
+                fields: vec![
+                    IdlField {
+                        name: "ret1".to_string(),
+                        docs: None,
+                        ty: IdlType::U64,
+                    },
+                    IdlField {
+                        name: "ret2".to_string(),
+                        docs: None,
+                        ty: IdlType::U64,
+                    }
+                ],
+            },
+        }
+    );
+}
+
+#[test]
+fn mangled_names() {
+    let src = r#"
+contract Testing {
+    function getNum(uint64 a, uint64 b) public pure returns (uint64 ret1, uint64 ret2) {
+        ret1 = a + b;
+        ret2 = a/b;
+    }
+
+    function getNum(int32 a, int32 b) public pure returns (int32 ret3, int32 ret4) {
+        ret3 = a-b;
+        ret4 = b/a;
+    }
+}
+    "#;
+
+    let ns = generate_namespace(src);
+    let idl = generate_anchor_idl(0, &ns);
+
+    assert_eq!(idl.instructions.len(), 3);
+
+    assert_eq!(idl.instructions[0].name, "new");
+
+    assert_eq!(idl.instructions[1].name, "getNum_uint64_uint64");
+    assert!(idl.instructions[1].docs.is_none());
+    assert!(idl.instructions[1].accounts.is_empty());
+    assert_eq!(idl.instructions[1].args.len(), 2);
+    assert_eq!(
+        idl.instructions[1].args[0],
+        IdlField {
+            name: "a".to_string(),
+            docs: None,
+            ty: IdlType::U64
+        }
+    );
+    assert_eq!(
+        idl.instructions[1].args[1],
+        IdlField {
+            name: "b".to_string(),
+            docs: None,
+            ty: IdlType::U64
+        }
+    );
+    assert_eq!(
+        idl.instructions[1].returns,
+        Some(IdlType::Defined("getNum_uint64_uint64_returns".to_string()))
+    );
+
+    assert_eq!(idl.instructions[2].name, "getNum_int32_int32");
+    assert!(idl.instructions[2].docs.is_none());
+    assert!(idl.instructions[2].accounts.is_empty());
+
+    assert_eq!(idl.instructions[2].args.len(), 2);
+    assert_eq!(
+        idl.instructions[2].args[0],
+        IdlField {
+            name: "a".to_string(),
+            docs: None,
+            ty: IdlType::I32,
+        }
+    );
+    assert_eq!(
+        idl.instructions[2].args[1],
+        IdlField {
+            name: "b".to_string(),
+            docs: None,
+            ty: IdlType::I32
+        }
+    );
+    assert_eq!(
+        idl.instructions[2].returns,
+        Some(IdlType::Defined("getNum_int32_int32_returns".to_string()))
+    );
+
+    assert_eq!(idl.types.len(), 2);
+
+    assert_eq!(
+        idl.types[0],
+        IdlTypeDefinition {
```

Comment:
Nit: That way these tests are hard to audit. Introducing `let expected_... = ` bindings would help